pkg/morph/client/netmap/config.go

```diff
@@ -82,42 +175,145 @@ func (s *SetConfigPrm) SetValue(value interface{}) {
+			configListMethod, err)
+	}
+
+	if ln := len(items); ln != 1 {
+		return fmt.Errorf("unexpected stack item count (%s): %d", configListMethod, ln)
+	}
+
+	arr, err := client.ArrayFromStackItem(items[0])
+	if err != nil {
+		return fmt.Errorf("record list (%s): %w", configListMethod, err)
+	}
+
+	return iterateRecords(arr, func(key, value []byte) error {
+		return f(key, value)
+	})
 }
 
-func StringAssert(item stackitem.Item) (interface{}, error) {
-	return client.StringFromStackItem(item)
+// ConfigWriter is an interface of NeoFS network config writer.
+type ConfigWriter interface {
+	UnknownParameter(string, []byte)
+	MaxObjectSize(uint64)
+	BasicIncomeRate(uint64)
+	AuditFee(uint64)
+	EpochDuration(uint64)
+	ContainerFee(uint64)
+	ContainerAliasFee(uint64)
+	EigenTrustIterations(uint64)
+	EigenTrustAlpha(float64)
+	InnerRingCandidateFee(uint64)
+	WithdrawFee(uint64)
+}
+
+// WriteConfig writes NeoFS network configuration received via iterator.
+//
+// Returns iterator's errors directly.
+func WriteConfig(dst ConfigWriter, iterator func(func(key, val []byte) error) error) error {
+	return iterator(func(key, val []byte) error {
+		switch k := string(key); k {
+		default:
+			dst.UnknownParameter(k, val)
+		case maxObjectSizeConfig:
+			dst.MaxObjectSize(bigint.FromBytes(val).Uint64())
+		case basicIncomeRateConfig:
+			dst.BasicIncomeRate(bigint.FromBytes(val).Uint64())
+		case auditFeeConfig:
+			dst.AuditFee(bigint.FromBytes(val).Uint64())
+		case epochDurationConfig:
+			dst.EpochDuration(bigint.FromBytes(val).Uint64())
+		case containerFeeConfig:
+			dst.ContainerFee(bigint.FromBytes(val).Uint64())
+		case containerAliasFeeConfig:
+			dst.ContainerAliasFee(bigint.FromBytes(val).Uint64())
+		case etIterationsConfig:
+			dst.EigenTrustIterations(bigint.FromBytes(val).Uint64())
+		case etAlphaConfig:
+			v, err := strconv.ParseFloat(string(val), 64)
+			if err != nil {
+				return fmt.Errorf("prm %s: %v", etAlphaConfig, err)
+			}
+
+			dst.EigenTrustAlpha(v)
+		case irCandidateFeeConfig:
+			dst.InnerRingCandidateFee(bigint.FromBytes(val).Uint64())
+		case withdrawFeeConfig:
+			dst.WithdrawFee(bigint.FromBytes(val).Uint64())
+		}
+
+		return nil
+	})
 }
 
-// ListConfigArgs groups the arguments
-// of config listing test invoke call.
-type ListConfigArgs struct {
+// ConfigPrm groups the arguments
+// of get config value test invoke call.
+type ConfigPrm struct {
+	key []byte
 }
 
-// ListConfigValues groups the stack parameters
-// returned by config listing test invoke.
-type ListConfigValues struct {
-	rs []stackitem.Item
+// SetKey sets binary key to configuration parameter.
+func (c *ConfigPrm) SetKey(v []byte) {
+	c.key = v
 }
 
-// IterateRecords iterates over all config records and passes them to f.
+// Config performs the test invoke of get config value
+// method of NeoFS Netmap contract.
+func (c *Client) Config(p ConfigPrm, assert func(stackitem.Item) (interface{}, error)) (interface{}, error) {
```

Comment:
do we need it public if we have getters for all (all, yes?) the config values?

Reply:
No, fixed.